File: LeetCode Solutions/March 2021 LeetCoding Challenge/Short Encoding of Words/Short Encoding of Words.cpp
class Solution {
public:
    int minimumLengthEncoding(vector<string>& words) {
        unordered_set<string> wordsSet(words.begin(), words.end());
        for (string word : words){
            if (wordsSet.find(word) != wordsSet.end()){
                for (int i=1; i<word.length(); i++){ 
                    wordsSet.erase(word.substr(i));
                }
            }
        }
        int lengthOfTheShortestReferenceString = wordsSet.size();
        for (string word : wordsSet){
            lengthOfTheShortestReferenceString += word.size();
        }
        return lengthOfTheShortestReferenceString;
    }
};

File: LeetCode Solutions/March 2021 LeetCoding Challenge/Short Encoding of Words/Short Encoding of Words.java
class Solution {
    public int minimumLengthEncoding(String[] words) {
        Set<String> wordsSet = new HashSet<>(Arrays.asList(words));
        for (String word : words){
            if (wordsSet.contains(word)){
                for (int i=1; i<word.length(); i++){ 
                    wordsSet.remove(word.substring(i));
                }
            }
        }
        int lengthOfTheShortestReferenceString = wordsSet.size();
        for (String word : wordsSet){
            lengthOfTheShortestReferenceString += word.length();
        }
        return lengthOfTheShortestReferenceString;
    }
}